src/main/java/com/linagora/james/mailets/GuessClassificationMailet.java

```diff
@@ -200,12 +212,28 @@ private String asJson(Mail mail) throws MessagingException, IOException {
         return new ClassificationRequestBodySerializer(mail, uuidGenerator).toJsonAsString();
     }
 
-    @VisibleForTesting void addHeader(Mail mail, String classificationGuess) {
-        try {
-            MimeMessage message = mail.getMessage();
-            message.addHeader(headerName, classificationGuess);
-        } catch (MessagingException e) {
-            LOGGER.error("Error occured while adding classification guess header", e);
-        }
+    @VisibleForTesting void addHeaders(Mail mail, String classificationGuesses) {
+        Function<String, Map<String, ClassificationGuess>> jsonToClassificationGuess = Throwing
+            .<String, Map<String, ClassificationGuess>>function(
+                s -> objectMapper.readValue(s, typeReference))
+            .fallbackTo(s -> {
+                LOGGER.error("Error occurred while deserializing classification guess");
+                return ImmutableMap.of();
+            });
+
+        Consumer<Map.Entry<String, ClassificationGuess>> addRecipientHeader = Throwing
+            .<Map.Entry<String, ClassificationGuess>>consumer(entry -> mail.addSpecificHeaderForRecipient(
+                PerRecipientHeaders.Header.builder()
+                    .name(headerName)
+                    .value(objectMapper.writeValueAsString(entry.getValue()))
+                    .build(),
+                new MailAddress(entry.getKey())))
+            .orTryWith(e -> LOGGER.error("Failed serializing " + headerName + " for " + e));
```

Comment:
wouldn't it be more readable if you extract these 2 functions to native java functions?